eZ/Publish/Core/Limitation/ObjectStateLimitationType.php

```diff
@@ -236,8 +236,49 @@ public function getCriterion(APILimitationValue $value, APIUserReference $curren
             return new Criterion\ObjectStateId($value->limitationValues[0]);
         }
 
-        // several limitation values: IN operation
-        return new Criterion\ObjectStateId($value->limitationValues);
+        $groupedLimitationValues = $this->groupLimitationValues($value->limitationValues);
+
+        if (count($groupedLimitationValues) === 1) {
+            // several limitation values: IN operation
+            return new Criterion\ObjectStateId($value->limitationValues);
```

Comment:
It should work the same regardless if we use `$value->limitationValues` or `$groupedLimitationValues[0]`. This `if` basically means: if there is only one group found, one ObjectStateId Criterion with all the limitation values is enough (because the IN operation will be performed on them).